spock-core/src/main/java/org/spockframework/runtime/ParameterizedSpecRunner.java

```diff
@@ -132,20 +136,45 @@ private int estimateNumIterations(Object[] dataProviders) {
     return result == Integer.MAX_VALUE ? -1 : result;
   }
 
-  private void runIterations(Iterator[] iterators, int estimatedNumIterations) {
-    if (runStatus != OK) return;
+  private void runIterations(final FeatureInfo currentFeature, final Iterator[] iterators, final int estimatedNumIterations) {
+    if (runStatus.get() != OK) return;
+
+    boolean atLeastOneIteration = false;
+
+    Scheduler schedulerForRunIterations = this.scheduler.deriveScheduler(!currentFeature.isReportIterations() || !currentFeature.isSupportParallelExecution());
+
+    while (haveNext(currentFeature, iterators)) {
+      atLeastOneIteration = true;
+      final Object[] dataValues = nextArgs(currentFeature, iterators);
 
-    while (haveNext(iterators)) {
-      initializeAndRunIteration(nextArgs(iterators), estimatedNumIterations);
+      if (haveNext(currentFeature, iterators)) {
+        schedulerForRunIterations.schedule(new Runnable() {
+          @Override
+          public void run() {
+            runStatus.set(OK);
```

Comment:
I overlooked that it is a `ThreadLocal` now. However, the fact remains that the `runStatus` of the iterations is now ignored. Before it would check `if (resetStatus(ITERATION) != OK) break;` and halt the following iterations now the result of `resetStatus` is simply ignored.